### PR TITLE
Add support for MBF21 weapon flags

### PR DESCRIPTION
**Pull request by XaserAcheron**
Adds support for MBF21's new weapon flags field -- works just the same as thing flags: +FLAGNAME to set, -FLAGNAME to unset.

The DEHWeaponFlag implementation is more complex than I'd have liked, but  I needed to iterate through the list a couple of times so a bunch of `static final const`s wouldn't cut it. Lemme know if you've got any ideas on how to slim this down a bit (or if it's OK as-is) -- my Java-Fu is weak. :P

### CODE REVIEW

File: src/main/java/net/mtrop/doom/tools/decohack/DecoHackParser.java [hunk 1]
```diff
@@ -2658,6 +2692,31 @@ private Integer matchFlagMnemonic()
...
+		String mnemonic = currentToken().getLexeme().toLowerCase();
+
+		for (int i = 0; i < DEHWeaponFlag.VALUES.length; i++)
+		{
+			if(DEHWeaponFlag.VALUES[i].getMnemonic().equals(mnemonic))
+			{
+				out = DEHWeaponFlag.VALUES[i].getValue();
+				break;
+			}
+		}
```
Comment (MTrop) [inline src/main/java/net/mtrop/doom/tools/decohack/DecoHackParser.java:2714]: There's definitely an improvement that can be done here by making a static name lookup map in `DEHWeaponFlag` but I'm fine with kicking that can down the road when I dig back into the parser, later.

File: src/main/java/net/mtrop/doom/tools/decohack/data/DEHWeapon.java [hunk 2]
```diff
@@ -334,6 +363,27 @@ public void writeObject(Writer writer, DEHWeapon weapon) throws IOException
...
+				{
+					if(more)
+						writer.append(" | ");
+					writer.append(DEHWeaponFlag.VALUES[i].getMnemonic());
+					more = true;
+				}
+			}
+
+			writer.append("\r\n");
+		}
```
Comment (MTrop) [inline src/main/java/net/mtrop/doom/tools/decohack/data/DEHWeapon.java:385]: In light of the counter-proposed changes to the field spec, this should be changed to write just the integer value a la `DEHThing`.

**Review by MTrop — CHANGES_REQUESTED** [review]
Good stuff. Just one required change plus an optional one that would take a little more time to change. Up to you, Xaser.

**Comment by XaserAcheron** [comment]
Simplified the export so it just writes the number now; works great in latest dsda-doom `master`, so all is well now. :)

I'll hold off on the other suggested change if you don't mind, 'cause I gotta shift my focus back to the port itself. I'm already juggling too many to-dos already. :P

**Review by MTrop — APPROVED** [review]
Cool. Merging it in.

**Pull request merged**